test/jasmine/tests/download_test.js

```diff
@@ -59,8 +70,55 @@ describe('Plotly.downloadImage', function() {
     it('should create link, remove link, accept options', function(done) {
         downloadTest(gd, 'svg', done);
     }, LONG_TIMEOUT_INTERVAL);
-});
 
+    it('should produce the right SVG output in IE', function(done) {
+        // mock up IE behavior
+        Lib.isIE = function() { return true; };
+        slzProto.serializeToString = function() {
+            return serializeToString.apply(this, arguments)
+                .replace(/(\(#)([^")]*)(\))/gi, '(\"#$2\")');
+        };
+        var savedBlob;
+        navigator.msSaveBlob = function(blob) { savedBlob = blob; };
```

Comment:
It would be best to use jasmine [`spyOn` and `callFake`](https://jasmine.github.io/2.0/introduction.html#section-Spies:_<code>and.callFake</code>) that way we wouldn't have to worry about restoring the unmocked behavior.

Reply:
ha, you can only `spyOn` something that already exists - so `msSaveBlob` I have to handle manually. But `Lib.isIE` and `serializeToString` (and perhaps `document.createElement`??) I can switch to `spyOn().and.callFake`

Reply:
Looks good!